test/integration/basic/test/error-recovery.js

```diff
@@ -407,5 +411,17 @@ export default (context, renderViaHTTP) => {
         }
       }
     })
+
+    it('should show valid error when thrown in _app getInitialProps', async () => {
```

Comment:
I think this test could be in the `app-document` suite. Also tip:

```
const errMsg = 'have an error from _app getInitialProps'
      _app.replace('// throw _app GIP err here', `static async getInitialProps() {
        throw new Error("${errMsg}")
      }`)
try {
      await check(
        () => renderViaHTTP('/'),
        /have an error from _app getInitialProps/
      )
} finally {
     await _app.restore()
}
```

By adding `finally` you ensure it's always restored back even when the test throws 💯

Reply:
That helped thanks! 🙏